BUG: Allow np.interp([], [], []) to return an empty array instead of error (GH-30316)

Sujit-Hiwale opened the pull request:
BUG: Fix np.interp([], [], []) returning ValueError (GH-30316)

When both `x` and `xp` are empty, np.interp previously raised
ValueError("array of sample points is empty"). This behavior is overly
strict, as empty input should naturally result in an empty output array
with the same shape as `x`.

This commit implements special handling for the case xp == [] and
x == [], returning an empty output array, while preserving all existing
error conditions:

* xp == [] and x non-empty → still raises ValueError
* len(fp) != len(xp) → still raises ValueError
* xp non-empty → normal interpolation

Tests are added to verify the corrected behavior.
Fixes GH-30316.

jorenham committed:
TYP: ``_core._umath_tests`` module stubs (#30327)

mattip reviewed (commented):
I think only the case `np.interp([], [], [])` should return empty. All others should be errors.

Sujit-Hiwale commented:
Thanks @mattip for the feedback.

Per your comments:

- Removed all additional tests that duplicate test_exceptions

- Removed the incorrect test allowing np.interp([], [], [1,2])
 
- Kept only the intended new behavior:
  np.interp([], [], []) returns an empty array
 
- Updated logic in compiled_base.c accordingly

Let me know if any further adjustments are needed!

mattip commented on numpy/_core/meson.build:
```diff
@@ -927,7 +927,7 @@ foreach gen_mtargets : [
     src_file.process('src/umath/loops_arithm_fp.dispatch.c.src'),
     [
-      X86_V3, X86_V2, 
+      X86_V3, X86_V2,
```
Please avoid changes not connected to this PR.

mattip commented on numpy/lib/tests/test_function_base.py:
```diff
@@ -3087,6 +3087,12 @@ def test_exceptions(self):
         assert_raises(ValueError, interp, 0, [0], [1, 2], period=360)
+        
+    def test_empty_triple_case(self):
+        # New expected behavior: np.interp([], [], []) returns []
```
```suggestion
```
